elrond-wasm/src/esdt.rs

```diff
@@ -0,0 +1,169 @@
+use hex_literal::hex;
+
+use crate::{
+	api::BigUintApi,
+	types::{Address, BoxedBytes, ContractCall, TokenIdentifier},
+};
+
+/// Address of the system smart contract that manages ESDT.
+/// Bech32: erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u
+pub const ESDT_SYSTEM_SC_ADDRESS_ARRAY: [u8; 32] =
+	hex!("000000000000000000010000000000000000000000000000000000000002ffff");
+
+pub fn esdt_system_sc_address() -> Address {
+	Address::from(ESDT_SYSTEM_SC_ADDRESS_ARRAY)
+}
+
+/// Proxy for the ESDT system smart contract.
+/// Unlike other contract proxies, this one has a fixed address,
+/// so the proxy object doesn't really contain any data, it is more of a placeholder.
+pub struct ESDTSystemSmartContractProxy<BigUint: BigUintApi> {
+	_phantom: core::marker::PhantomData<BigUint>,
+}
+
+impl<BigUint: BigUintApi> ESDTSystemSmartContractProxy<BigUint> {
+	pub fn new() -> Self {
+		ESDTSystemSmartContractProxy {
+			_phantom: core::marker::PhantomData,
+		}
+	}
+
+	/// Produces a contract call to the ESDT system SC,
+	/// which causes it to issue a new ESDT token.
+	pub fn issue(
+		&self,
+		issue_cost: BigUint,
```

Comment:
Since the issue cost is always 5 eGLD, maybe have it as a constant in this file?

Reply:
The issue cost will almost certainly not stay 5 EGLD. I would actually remove the number from the contract altogether and let the protocol validate. Otherwise, a protocol update might leave the contract broken

Reply:
If we leave it up to the protocol to do the validation - and the validation fails, are the funds returned to the smart contract or to the user which triggered the call to the issue?

Reply:
good point, we need to check